tools/ScenarioTest.ResourceManager/PermissiveRecordMatcherWithApiExclusion.cs

```diff
@@ -116,22 +147,69 @@ protected bool ContainsIgnoredProvider(string requestUri, out string version)
                 !requestUri.Contains("/servicePrincipals?") &&
                 !requestUri.StartsWith("/webhdfs/v1/?aclspec", StringComparison.InvariantCultureIgnoreCase))
             {
-                version = String.Empty;
+                apiVersion = String.Empty;
                 return true;
             }
 
-            foreach (var provider in _providersToIgnore)
+            // Ignore resource providers
+            foreach (var provider in providersToIgnore)
             {
                 var providerString = string.Format("providers/{0}", provider.Key);
                 if (requestUri.Contains(providerString))
                 {
-                    version = provider.Value;
+                    apiVersion = provider.Value;
                     return true;
                 }
             }
 
+            if (resourcesToIgnore != null && resourcesToIgnore.Any())
+            {
+                // If we're looking at a specific provider and we have top level resource from this provider to ignore
+                foreach (var resourceToIgnore in resourcesToIgnore)
+                {
+                    var segments = requestUri.Split('/');
```

Comment:
@johnpaulkee you should add `StringSplitOptions.RemoveEmptyEntries` in this `.Split()` call to remove the empty segments

_Note_: you'll have to shift some numbers below since the first entry is now removed